datafusion/core/src/physical_optimizer/optimizer.rs

```diff
@@ -124,6 +125,10 @@ impl PhysicalOptimizer {
             // are not present, the load of executors such as join or union will be
             // reduced by narrowing their input tables.
             Arc::new(ProjectionPushdown::new()),
+            // SanityChecker rule checks whether the order and the
+            // partition requirements of each execution plan is
+            // satisfied by its children or not.
+            Arc::new(SanityCheckPlan::new()),
```

Comment:
We can discuss if this rule can be merged with `PipelineChecker`.

Reply:
From my understanding, in addition to doing `is_pipeline_friendly`, `PipelineChecker` also checks a condition around `SymmetricHashJoin` operation. Since `PipelineChecker` is more enchanced (to check pipeline friendliness) should we remove the pipeline friendliness check from `SanityChecker`? Or to simplify, I can merge `PipelineChecker` into `SanityChecker` (and remove `PipelineChecker`).

Reply:
I think, we should either 
- remove pipeline friendliness check from `SanityChecker` (as this is done in another rule already)
- or merge `PipelineChecker` into `SanityChecker` .
However, merging these two rules seems more appropriate. Hence, we should follow this approach.